### ISSUE TITLE
Stand-alone onboarding site, part I - installing the CLI

### ISSUE THREAD

**Issue by kytrinyx** [exercism/exercism.io#2047]
I've been giving onboarding a lot of thought. It's extraordinarily difficult to use exercism, and in particular, it's really, really difficult to get started.

Some people have been doing a lot to help out in this department, notably @caseywatts who is doing user research (https://github.com/exercism/website/issues/14), and @meaganewaller who helped put together a progress/flow on the dashboard.

I was reading an article this morning entitled [Give Your User's a Taste of Pure Badassery](https://medium.com/@allanbreyes/give-your-users-a-taste-of-pure-badassery-b1a8ad4a3da9) by @allanbreyes

It opens like this:

> “Wow! I love your app because I start out completely useless. I’m fumbling up this learning curve and I have no idea what I’ll be able to do once I spend the time to master it!” — no user ever

You could safely replace _your app_ with _exercism_ here, unfortunately.

This is also about working toward the "Minimum Badass User", as Kathy Sierra [talks about in this video](http://vimeo.com/54469442).

What if we had a little, stand-alone angular (or something) app, that

1. Let you choose your platform, or helped you figure out which platform you even have.
2. Guided you through the process of downloading and installing the CLI.
3. Let you choose a language.
4. told you to `exercism fetch tutorial <language>` or something
5. guided you through the process of TDD using a really simple problem that isn't about solving a problem, but only about "getting" the cycle of seeing a test failure and changing the message.

[UPDATE: Step 5 would probably be a separate, full tutorial, possibly in a different stand-alone app]

People could skip right past all of this if they wished, because it wouldn't be part of the main site, their dashboard would just have an invitation to try it, and it would be linked to from the help docs.

By making it completely stand-alone we don't have to rely on any of the existing code (most of which is far from exemplary), and people could contribute to this without knowing anything about other parts of the site/system.

Thoughts?

**Comment by kytrinyx** [comment]
Nice, thanks!

**Comment by kytrinyx** [comment]
This is trying to address two things, I think: on the one hand it's trying to make it easier to understand how exercism works (which is crucial, I think) and on the other hand it's trying to make it easier for beginners to get started with exercism.

This last topic is a huge and complex topic, and a lot of different suggestions have come up over time. I've collected what I found into one high-level topic about lowering the barrier to entry.

I've come to realize that, while important, it's not the first thing to work on. The most important thing to figure out on exercism is how to ensure that the conversations are rich and useful, and that everyone who submits code gets good, high quality feedback.

Once that problem is solved to a useful degree we can come back to figuring out how to get new programmers onboarded more easily.

Thanks everyone, for discussing!
